### ISSUE TITLE
Functions not differentiating between NA and "NA"

### ISSUE THREAD

**Issue by ChrisMuir** [ChrisMuir/refinr#11]
Here's an example:
```r
vect <- c("cats", "CATS", "NA", "na", "na na", NA, NA_character_, " ", "")
n_gram_merge(vect)
#> [1] "CATS" "CATS" NA     NA     NA     NA     NA     " "    ""
```
The expected output is this:
```r
#> [1] "CATS" "CATS" "NA"   "NA"   "NA"   NA     NA     " "    ""
```
Where the string literal `"NA"` and the missing value `NA` are kept separate, and not merged together.

**Comment by ChrisMuir** [comment]
Fixed in commit [6f2cbe7](https://github.com/ChrisMuir/refinr/commit/6f2cbe7cd5e4a4d7faceba52f402a2a53203a41b).
```r
vect <- c("cats", "CATS", "NA", "na", "na na", NA, NA_character_, " ", "")

library(refinr)

key_collision_merge(vect)
#> [1] "CATS" "CATS" "NA"   "NA"   "NA"   NA     NA     " "    ""

n_gram_merge(vect)
#> [1] "CATS" "CATS" "NA"   "NA"   "NA"   NA     NA     " "    ""
```

**Comment by ChrisMuir** [comment]
Found another bug related to this issue, introduced in the latest commit [0db081f](https://github.com/ChrisMuir/refinr/commit/0db081f85d3965dbfc8e21f76dea925873959db7):
```r
vect <- c("cats", "CATS", "NA", "na", "na na", NA, NA_character_, " ", "")

library(refinr)

n_gram_merge(vect)
#> "CATS" "CATS" "NA"   " "    " "    NA     NA     " "    ""
```
